### PR TITLE
Feature/devise

### PR DESCRIPTION
**Pull request by Morrism1**
# MEMBERS-ONLY

> In this project, we created the:

* added validations, 

* generated migrations

* created associations between models, and 

* added user authentication using Devise.

* styled components using bootstrap

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello:wave: @Morrism1, @mugberto. Good work here... Impressive interface:+1:. Here are your required improvements:**

:point_right: You have used the devise _authenticate\_user!_ helper method to restrict access to post's new and create actions :+1:. Now, you should create a custom method in Application Controller that mimics the _authenticate\_user!_ and use this method instead. The method should verify that a user is signed in and if not, redirect to sign in page.

:point_right: In posts index, you have used devise _user_\signed\_in?_ method to check if a user is signed in :+1: Now create a custom method in Application Controller to mimic this behaviour and use this custom method instead.


:point_right: Remove all conditionals/logic from the posts index view. Extract them into helper methods. 



:point_right: Apart from the devise registration feature, implement your own custom user sign up feature. Set up the controller actions (new and create), views and routes required for this to work.

:point_right: _**Optional:**_ To make application controller methods available in your views, you can declare them as helper methods like shown below:

```ruby
helper_method :signed_in?
def signed_in?
  !current_user.nil?
end
```

_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested:exclamation:

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  devise_for :users, :controllers => {registrations: "registrations"}
+  resources :posts
```
Comment (maelfosso) [inline config/routes.rb]: The routes are restricted to some actions as required [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png)

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

You did a really really good job :+1: 
Nothing to say, except about the routes restrictions as indicated in the comments.
Kindly update and submit a new code review request

Thanks

Mael FOSSO

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**